scripts/build/setup/termux_setup_cmake.sh

```diff
@@ -5,15 +5,26 @@ termux_setup_cmake() {
 	local TERMUX_CMAKE_TARNAME=cmake-${TERMUX_CMAKE_VERSION}-Linux-x86_64.tar.gz
 	local TERMUX_CMAKE_TARFILE=$TERMUX_PKG_TMPDIR/$TERMUX_CMAKE_TARNAME
 	local TERMUX_CMAKE_FOLDER=$TERMUX_COMMON_CACHEDIR/cmake-$TERMUX_CMAKE_VERSION
-	if [ ! -d "$TERMUX_CMAKE_FOLDER" ]; then
-		termux_download https://cmake.org/files/v$TERMUX_CMAKE_MAJORVESION/$TERMUX_CMAKE_TARNAME \
+
+	if [ -z "$TERMUX_ON_DEVICE_BUILD" ]; then
+		if [ ! -d "$TERMUX_CMAKE_FOLDER" ]; then
+			termux_download https://cmake.org/files/v$TERMUX_CMAKE_MAJORVESION/$TERMUX_CMAKE_TARNAME \
 				"$TERMUX_CMAKE_TARFILE" \
 				757e7ddd29d11422334a60d7cd9c19b7af20349a03cb15aa3c879f1b54193c95
-		rm -Rf "$TERMUX_PKG_TMPDIR/cmake-${TERMUX_CMAKE_VERSION}-Linux-x86_64"
-		tar xf "$TERMUX_CMAKE_TARFILE" -C "$TERMUX_PKG_TMPDIR"
-		mv "$TERMUX_PKG_TMPDIR/cmake-${TERMUX_CMAKE_VERSION}-Linux-x86_64" \
-		   "$TERMUX_CMAKE_FOLDER"
+			rm -Rf "$TERMUX_PKG_TMPDIR/cmake-${TERMUX_CMAKE_VERSION}-Linux-x86_64"
+			tar xf "$TERMUX_CMAKE_TARFILE" -C "$TERMUX_PKG_TMPDIR"
+			mv "$TERMUX_PKG_TMPDIR/cmake-${TERMUX_CMAKE_VERSION}-Linux-x86_64" \
+				"$TERMUX_CMAKE_FOLDER"
+		fi
+
+		export PATH=$TERMUX_CMAKE_FOLDER/bin:$PATH
+	else
+		if [ "$(dpkg-query -W -f '${db:Status-Status}\n' cmake 2>/dev/null)" != "installed" ]; then
+			echo "Package 'cmake' is not installed."
+			echo "Install it with command 'pkg install cmake'."
```

Comment:
Small nit, "Build it first with 'build-package.sh cmake' or install it...", same for messages below.

Reply:
> same for messages below

For rust too ? That package takes 6 hours to build on my PC, don't know how much it will take on device. It also requires that LLVM is built from source. And everything without any error.

I'll check other cases to see which packages are can be built on device, but rust is definetly can be only installed for now.

Reply:
Sure, if you know that, but I think we can give both the build and prebuilt install options for the rest. Btw, I built Swift from scratch several times on my 2017 flagship device, takes me 3.5 hours, including their forked LLVM and clang. The rust compiler may be slower though. :wink:

Reply:
Done.

Cmake, ninja & protobuf are compiled on device without any problem.
Golang should be possible too - I did that without termux-packages environment previously. Though it may require some changes to build.sh.